src/components/rangeslider/index.js

```diff
@@ -0,0 +1,292 @@
+            var delta = +e.clientX - startX;
+
+            switch(target) {
+                case slideBox:
+                    slider.style.cursor = 'ew-resize';
+                    setPixelRange(+maxVal + delta, +minVal + delta);
+                    break;
+
+                case grabAreaMin:
+                    slider.style.cursor = 'col-resize';
+                    setPixelRange(+minVal + delta, +maxVal);
+                    break;
+
+                case grabAreaMax:
+                    slider.style.cursor = 'col-resize';
+                    setPixelRange(+minVal, +maxVal + delta);
+                    break;
+
+                default:
+                    slider.style.cursor = 'ew-resize';
+                    setPixelRange(offsetX, offsetX + delta);
+                    break;
+            }
+        }
+
+        function mouseUp() {
+            window.removeEventListener('mousemove', mouseMove);
+            window.removeEventListener('mouseup', mouseUp);
+            slider.style.cursor = 'auto';
+        }
+    });
+
+
+    function setRange(min, max) {
+        var rangeMin = fullLayout.xaxis.range[0],
+            rangeMax = fullLayout.xaxis.range[1],
+            range = rangeMax - rangeMin,
+            pixelMin = (min - rangeMin) / range * width,
+            pixelMax = (max - rangeMin) / range * width;
+
+        setPixelRange(pixelMin, pixelMax);
+    }
+
+
+    function setPixelRange(min, max) {
+
+        min = Lib.constrain(min, 0, width);
+        max = Lib.constrain(max, 0, width);
+
+        if(max < min) {
+            var temp = max;
+            max = min;
+            min = temp;
+        }
+
+        helpers.setAttributes(slider, {
+            'data-min': min,
+            'data-max': max
+        });
+
+        helpers.setAttributes(slideBox, {
+            'x': min,
+            'width': max - min
+        });
+
+        helpers.setAttributes(maskMin, { 'width': min });
+        helpers.setAttributes(maskMax, {
+            'x': max,
+            'width': width - max
+        });
+
+        helpers.setAttributes(grabberMin, { 'transform': 'translate(' + (min - handleWidth - 1) + ')' });
+        helpers.setAttributes(grabberMax, { 'transform': 'translate(' + max + ')' });
+
+        // call to set range on plot here
+        var rangeMin = fullLayout.xaxis.range[0],
+            rangeMax = fullLayout.xaxis.range[1],
+            range = rangeMax - rangeMin,
+            dataMin = min / width * range + rangeMin,
+            dataMax = max / width * range + rangeMin;
+
+        Plotly.relayout(gd, 'xaxis.range', [dataMin, dataMax]);
+    }
+
+
+    var rangePlots = rangePlot(gd, width, height);
+
+    helpers.appendChildren(slider, [
+        sliderBg,
+        rangePlots,
+        maskMin,
+        maskMax,
+        slideBox,
+        grabberMin,
+        grabberMax
+    ]);
+
+    var infoLayer = fullLayout._infolayer;
+    infoLayer.selectAll('g.range-slider')
+        .data([0])
```

Comment:
So there can only be one range slider on the plot at the same time?

It would be nice to generalize so that each xaxis could have its own range slider.